doc/user_guide/psyir.rst

```diff
@@ -61,74 +61,53 @@ PSyIR Nodes
-subclassed into many directives associated with OpenMP and
-OpenACC. The ``Kern`` class is subclassed into ``CodedKern``,
-``InlinedKern`` and ``BuiltinKern``.
-
-
-Kernel-layer nodes
-------------------
-
-Kernel-layer PSyIR classes are currently used to describe existing
-code in a language independent way. Consequently these nodes are more
-prescriptive and are independent of a particular PSyclone
-frontend. These nodes are designed to be used with PSyIR backends. Two
-PSy-layer classes (``Loop`` and ``Schedule``) can also be used as
-Kernel-layer classes. Additionally, the ``Schedule`` class is further
-subclassed into a ``Routine`` and then a kernel-layer
-``KernelSchedule``.  In addition to ``KernelSchedule``, Kernel-layer
-PSyIR nodes are: ``Loop``, ``WhileLoop``, ``IfBlock``, ``CodeBlock``,
-``Assignment``, ``Range``, ``Reference``, ``Operation``, ``Literal``, ``Call``,
-``Return`` and ``Container``. The ``Reference`` class is further
-subclassed into ``ArrayReference``, ``StructureReference`` and
-``ArrayOfStructuresReference``, the ``Operation`` class is further
-subclassed into ``UnaryOperation``, ``BinaryOperation`` and
-the ``Container`` class is further subclassed
-into ``FileContainer`` (representing a file that may contain more than
-one ``Container`` and/or ``Routine``. Those nodes representing
-references to structures (derived types in Fortran) have a ``Member``
-child node representing the member of the structure being
-accessed. The ``Member`` class is further subclassed into
-``StructureMember`` (representing a member of a structure that is
-itself a structure), ``ArrayMember`` (a member of a structure that is
-an array of primitive types) and ``ArrayOfStructuresMember`` (a member
-of a structure this is itself an array of structures).
-
-
-Node Descriptions
-=================
-
-The Range node
---------------
+types: language-level nodes, which are nodes that the PSyIR backends
+support, and therefore they can be directly translated to code; and
+higher-level nodes, which are additional nodes that each domain can
+insert and they need to implement a `lower_to_language_level` method
+in order to be converted to its equivalent representation using only
+language-level to be converted to code.
+
+The rest of this document describe the language-level nodes, but they
+all inherit from the same base-class, so the methods described here are
+applicable to all nodes.
+
+
+Available language-level nodes
+==============================
+
+- :ref_guide:`ArrayMember psyclone.psyir.nodes.array_member.html`
+- :ref_guide:`ArrayReference psyclone.psyir.nodes.array_reference.html`
+- :ref_guide:`ArrayOfStructuresMember psyclone.psyir.nodes.array_of_structures_member.html`
+- :ref_guide:`ArrayOfStructuresReference psyclone.psyir.nodes.array_of_structures_reference.html`
+- :ref_guide:`Assignment psyclone.psyir.nodes.assignment.html`
+- :ref_guide:`BinaryOperation psyclone.psyir.nodes.binary_operation.html`
```

Comment:
This causes a link-check failure. Probably because BinaryOperation is actually in operation.py? Same for unaryoperation, range, return and binaryoperation.

Reply:
Ah true, I changed the format to something that exists for all of them now.